### DISCUSSION TITLE
MWAA - CopyFromExternalStageToSnowflakeOperator - Failed with error "Invalid regular expression pattern in airflow"

### DISCUSSION THREAD

**Discussion by GomathiMa** [apache/airflow#33964] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Airflow Version:2.5.1

While using MWAA, CopyFromExternalStageToSnowflakeOperator: Failed with error "Invalid regular expression pattern in airflow"

Error
snowflake.connector.errors.ProgrammingError: 091003 (22000):

Note: I am trying to use {{ ds_nodash }} variable in pattern. This variable value is not getting replaced in the pattern while generating copy into statement. However variable getting printed correctly in echo and works well with Snowflake operator - COPY TO as sql.

### How to reproduce

Use CopyFromExternalStageToSnowflakeOperator and have pattern value contains the variable {{ ds_nodash }}

### Operating System

Windows

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
You have to check if the fields is in template_fields. If not you can extend the operator and add it to your template_fields. The easiest way Extend the operator:

https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/operators.html#concepts-jinja-templating

```
class MyOperator(OriginalOperator):
    templated_fields = () # <- copy fields from the original one and the one you are missing

```

And use MyOperator instead of the OriginalOperator

You can also likely do in your DAG:

```
OriginalOperator.templated_fields = () # <- here add the list of fields you want to be templated.
```

**Reply by Taragolis** [reply]
@GomathiMa, in additional you could propose changes into [operator templated fields](https://github.com/apache/airflow/blob/21e8f878a3c91250d0d198c6c3675b4b350fcb61/airflow/providers/snowflake/transfers/copy_into_snowflake.py#L67-L68) by submitting a PR, seems like no one attribute processed in the operator constructor so it should be easy changes, for example in the last week same changes in the other operator was approved pretty quick: https://github.com/apache/airflow/pull/33585

**Reply by angelarosario** [reply]
Thanks for this. 
For anyone else doing the second method, the correct attribute name is `template_fields`
`OriginalOperator.template_fields = () # <- here add the list of fields you want to be templated.`
